cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1748,6 +1751,16 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
                     });
         }
 
+        private Kafka updateAssemblyKafkaStorage(Kafka kafkaAssembly, KafkaCluster kafkaCluster) {
+            return new KafkaBuilder(kafkaAssembly)
+                .editSpec()
+                    .editKafka()
+                        .withStorage(kafkaCluster.getStorage())
+                    .endKafka()
+                .endSpec()
+                .build();
+        }
```

Comment:
You should not modify it like this. Certainly not here because that might screw up other things. You should use `kafkaCluster.getStorage()` where you need the storage.